test/jit/test_complex.py

```diff
@@ -55,3 +58,54 @@ def fn(a: int, b: torch.Tensor, dim: int):
         t2 = torch.tensor([0.4, 1.4j, 2.35])
 
         self.checkScript(fn, (t1, t2, 2))
+
+    def test_complex_math_ops(self):
+        vals = ([0.0, 1.0, 2.2, -1.0, -0.0, -2.2, 1, 0, 2]
+                + [10.0 ** i for i in range(5)] + [-(10.0 ** i) for i in range(5)])
+        complex_vals = tuple(complex(x, y) for x, y in product(vals, vals))
+
+        def checkMath(func_name):
+            funcs_template = dedent('''
+            def func(a: complex):
+                return cmath.{func}(a)
+            ''')
+
+            funcs_str = funcs_template.format(func=func_name)
+            scope = {}
+            execWrapper(funcs_str, globals(), scope)
+            cu = torch.jit.CompilationUnit(funcs_str)
+            f_script = cu.func
+            f = scope['func']
+
+            for a in complex_vals:
+                res_python = None
+                res_script = None
+                try:
+                    res_python = f(a)
+                except Exception as e:
+                    res_python = e
+                try:
+                    res_script = f_script(a)
+                except Exception as e:
+                    res_script = e
+
+                if res_python != res_script:
+                    if isinstance(res_python, Exception):
+                        continue
+
+                    msg = ("Failed on {func_name} with input {a}. Python: {res_python}, Script: {res_script}"
+                           .format(func_name=func_name, a=a, res_python=res_python, res_script=res_script))
+                    self.assertEqual(res_python, res_script, msg=msg)
+
+        unary_ops = ['log', 'log10', 'sqrt', 'exp', 'sin', 'cos', 'asin', 'acos', 'atan', 'sinh', 'cosh',
+                     'tanh', 'asinh', 'acosh', 'atanh', 'phase']
```

Comment:
I would still recommend generating these programmatically so that future ops are automatically tested but I won't not stamp because of it.